### PR TITLE
Fix for issue #441: Offline events should be fired once

### PR DESCRIPTION
**Pull request by anhldbk**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -32,10 +32,13 @@ function defaultId () {
 function sendPacket (client, packet, cb) {
   try {
     var buf = mqttPacket.generate(packet);
-    if (!client.stream.write(buf) && cb) {
-      client.stream.once('drain', cb);
-    } else if (cb) {
+
+    client.emit('packetsend', packet);
```
Comment (mcollina) [inline lib/client.js]: why are you emitting `'packetsend'` here?
Can you send another PR for `'packetsend'`  and `'packetreceive'`, so it's easier to discuss?

**Comment by mcollina** [comment]
Can you please add an explanation on the README of the various status flags? And maybe an explanation of the state machine?

**Comment by anhldbk** [comment]
@mcollina Ok, I'm gonna update the README.

Btw, you said: 
```text
why are you emitting 'packetsend' here?
Can you send another PR for 'packetsend'  and 'packetreceive', so it's easier to discuss?
```
Still I really don't know with the diff about `emitting packetsend`. I looked at the code [Current client.js, line #274](https://github.com/mqttjs/MQTT.js/blob/master/lib/client.js#L274) and found no difference.

**Comment by mcollina** [comment]
@anhldbk github diff messed things up, and that line showed up as added. I checked and I removed my comment.

**Comment by anhldbk** [comment]
@mcollina I updated the README. Is that ok?

**Comment by mcollina** [comment]
All good for me.

I think this is semver-minor. Anybody thinks it's semver-major?

**Comment by RangerMauve** [comment]
I think it's minor. I doubt many people would have used the "offline" event in a way where it needs to be called each time a reconnection is attempted.

**Comment by mcollina** [comment]
Ok, to be minor, merging and releasing.
